agent_deploy/kubernetes/sysdig-agent-daemonset-v2.yaml

```diff
@@ -117,9 +117,13 @@ spec:
           exec:
             command: [ "test", "-e", "/opt/draios/logs/running" ]
           initialDelaySeconds: 10
+        env:
+        - name: K8S_NODE
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
```

Comment:
For consistency, I think it is cleaner to write it down in the `podinfo` volume, as we are already doing for `namespace` and `name`. What's your opinion?

Reply:
Yes, I think it would be cleaner, but if we write in the `podinfo` volume we need to change the agent code as well to consume it, so it will be used starting from the first version including the change.

We could:
1. have the node name in both places for a transitory time.
2. do the change in agent to consume the file
3. decommission the env variable

Reply:
Considering that currently, K8S_NODE env is useless because no one is setting it, I wouldn't pay too much effort here. I would go just setting it in a file. BTW it's your call

Reply:
My intention was exactly to give a reason for the `K8S_NODE` env variable to exist without touching agent code :-), but I take your point about consistency. If I had to do it from scratch I would use env variables everywhere because they are easier to consume for non-structured values, but as it does not change much I will comply with what you suggest.
I will do the change in agent first, that I will modify this.

Reply:
Hold-on, I double-checked and the node name is only available through the environment variable:

> The following information is available through environment variables:
>
>status.podIP - the pod's IP address
>spec.serviceAccountName - the pod's service account name, available since v1.4.0-alpha.3
>spec.nodeName - the node's name, available since v1.4.0-alpha.3
>status.hostIP - the node's IP, available since v1.7.0-alpha.1

https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api

we have no options ;-)

Reply:
Approval arriving then